File: app-java/src/main/java/com/github/simonpham/hasbrain/eaj/ui/browser/BrowserActivity.java
package com.github.simonpham.hasbrain.eaj.ui.browser;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import com.github.simonpham.hasbrain.eaj.R;

/**
 * Created by Simon Pham on 5/26/18.
 * Email: simonpham.dn@gmail.com
 */
public class BrowserActivity extends AppCompatActivity {

    private WebView vWeb = null;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_browser);

        initialize();

        String url = getIntent().getStringExtra("MainActivity.ARTICLE_URL");
        loadUrl(url);
    }

    @SuppressLint("SetJavaScriptEnabled")
    private void initialize() {
        vWeb = findViewById(R.id.webView);

        // enable javascript
        vWeb.setWebViewClient(new WebViewClient());
        vWeb.getSettings().setJavaScriptEnabled(true);
    }

    private void loadUrl(String url) {
        if (vWeb != null) vWeb.loadUrl(url);
    }
}

File: app/src/main/java/com/github/simonpham/hasbrain/ea/ui/browser/BrowserActivity.kt
package com.github.simonpham.hasbrain.ea.ui.browser

import android.annotation.SuppressLint
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.webkit.WebViewClient
import com.github.simonpham.hasbrain.ea.R
import kotlinx.android.synthetic.main.activity_browser.*

class BrowserActivity : AppCompatActivity() {

    @SuppressLint("SetJavaScriptEnabled")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_browser)

        // enable Javascript
        webView.webViewClient = WebViewClient()
        webView.settings.javaScriptEnabled = true

        val url = intent.getStringExtra("MainActivity.ARTICLE_URL")
        loadUrl(url)
    }

    private fun loadUrl(url: String) {
        if (webView != null) webView.loadUrl(url)
    }
}
